### ISSUE TITLE
Test suite

### ISSUE THREAD

**Issue by wavesinaroom** [wavesinaroom/my_odin_project_synthesizer#13]
After spending a long time finding out how to override `audio.js` for testing, I came up with a way to mock that module.
However, that mock was simply beginning of the solution because I'm just able to test UI for the moment and I want to go deeper with my tests.
For that reason, I'm opening an issue dedicated to testing so I can focus completely on mocking and test performance throughout the synths modules.
In other words, I've planned to have two milestones for this test:

1. `audio.js` and potentially some features of Web Audio API mock
2. UI, `audio.js` values change, functions calls tests
